src/main/java/org/automation/utilities/WebdriverWaits.java

```diff
@@ -124,14 +122,66 @@ public static void fluentWait_ElementLocated(long waitTimeForTimeout, long waitT
         wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
 
     }
-    
+
     public static void fluentWait_ElementIntactable(long waitTimeForTimeout, long waitTimeForPolling, By locator) {
-        Wait<WebDriver> wait = new FluentWait<>(getDriver()).withTimeout(Duration.ofSeconds(waitTimeForTimeout))
+        Wait<WebDriver> wait = new FluentWait<>(getDriver())
+                .withTimeout(Duration.ofMillis(waitTimeForTimeout))
                 .pollingEvery(Duration.ofMillis(waitTimeForPolling))
+                .ignoring(NoSuchElementException.class)
                 .ignoring(ElementNotInteractableException.class);
         wait.until(ExpectedConditions.elementToBeClickable(locator));
+    }
+
+    public static void fluentWaitForDuration(long timeoutInMillis, long pollingInMillis) {
+        Wait<WebDriver> wait = new FluentWait<>(getDriver())
+                .withTimeout(Duration.ofMillis(timeoutInMillis))  // Total wait time
+                .pollingEvery(Duration.ofMillis(pollingInMillis)) // Polling interval
+                .ignoring(Exception.class); // Ignore exceptions during polling
+
+        // Wait until timeout is reached (dummy condition)
+        wait.until(driver -> true); // A condition that always returns false, making it wait the full duration
+    }
+
+    public static void waitForElementValueToUpdate(long timeoutInMillis, long pollingInMillis, By locator, String oldValue) {
+        Wait<WebDriver> wait = new FluentWait<>(getDriver())
+                .withTimeout(Duration.ofMillis(timeoutInMillis))
+                .pollingEvery(Duration.ofMillis(pollingInMillis))
+                .ignoring(Exception.class);
+
 
+        WebElement element = wait.until(driver -> {
+            WebElement ele = driver.findElement(locator);
+            String currentValue = ele.getAttribute("value");
+
+            if (!currentValue.equals(oldValue)) {
+                return ele;
+            } else {
+                return null;
+            }
+        });
     }
+
+    public static void retryClick(WebElement element, int maxRetries) {
```

Comment:
Improvement Suggestion: Add a delay between retries to allow the element to become interactable.